src/RestQuery.js

```diff
@@ -290,7 +298,11 @@ RestQuery.prototype.replaceSelect = function() {
       values.push(result[selectValue.key]);
     }
     delete selectObject['$select'];
-    selectObject['$in'] = values;
+    if (Array.isArray(selectObject['$in'])) {
```

Comment:
you ca simplify with selectObject['$in'] = [].concat(selectObject['$in'] || [], values);